internal/infrastructure/kubernetes/ratelimit/resource_provider.go

```diff
@@ -248,7 +248,7 @@ func (r *ResourceRender) Deployment() (*appsv1.Deployment, error) {
 					TerminationGracePeriodSeconds: ptr.To[int64](300),
 					DNSPolicy:                     corev1.DNSClusterFirst,
 					RestartPolicy:                 corev1.RestartPolicyAlways,
-					SchedulerName:                 "default-scheduler",
+					SchedulerName:                 r.rateLimitDeployment.Pod.SchedulerName,
```

Comment:
if it's empty, use `default-scheduler` as default value?

Reply:
That's already added by kubernetes when left empty. Wouldn't it be preferable to simply leave this field empty if not set by the user?

Reply:
If they mean the same, let's be consistent with before for compatibility, WDYT?

Reply:
will this change trigger a rollout of existing deployment/daemonset?

Reply:
If a user has a mutating hook to inject a custom scheduler, it can become problematic when gateway tries to reconcile this resource and inject something else. By leaving it empty that allows for a clean merge.

This change will not trigger a rollout as kubernetes is already adding if this field is empty. I've confirmed this locally as well by removing the scheduler field from a deployment and daemonset resource.

Reply:
if it will not trigger a rollout, lgtm.